### PR TITLE
Fixes  

### LINKED ISSUE #30: Corridor Query Fails When Attempting To Set Spatial Extent
QGIS 3.28.6 Plugin 13.3

When building a corridor query against the following service an error is triggered when the user clicks on the set spatial extent.

image

2024-05-03T10:31:05     WARNING    Python error : An error has occurred while executing Python code: See message log (Python Error) for more details.
2024-05-03T10:32:26     WARNING    Traceback (most recent call last):
              File "C:\Users/guy.collins/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\edr_plugin\gui\__init__.py", line 496, in set_query_extent
              data_query_tool_cls(self)
              File "C:\Users/guy.collins/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\edr_plugin\gui\query_tools.py", line 499, in __init__
              self.setup_data_query_tool()
              File "C:\Users/guy.collins/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\edr_plugin\gui\query_tools.py", line 518, in setup_data_query_tool
              width_units = corridor_query_data["link"]["variables"]["width_units"]
             KeyError: 'width_units'

This error seems to apply to many different collections on this service.

### LINKED ISSUE #32: Disable Z if the collection does not support it
If I set a trajectory and there is no Z dimension, editing the spatial extent results in adding Z to the request.

### LINKED ISSUE #33: Redesign the trajectory window
image

@Gugym to provide a sketch with tooltip and text to update the window for consistency and clarity.

### LINKED ISSUE #34: Trajectory data gets lost when changing collection
If you enter data in trajectory, go back to the main window and select another collection, the trajectory data gets wiped.

### PR DESCRIPTION
**Pull request by JanCaha**
Fixes #32 - the WKT will only contained dimensions supported by the collection
Fixes #30 - if metadata for corridor are missing, an error is show and user cannot build corridor query
Fixes #34 - if the newly selected collection supports the same type of query and same CRS
Fixes #33

### CODE REVIEW

File: edr_plugin/gui/ui/query_corridor.ui [hunk 1]
```diff
@@ -230,12 +247,12 @@
...
       <font>
-       <weight>75</weight>
-       <bold>true</bold>
+       <weight>50</weight>
+       <bold>false</bold>
       </font>
      </property>
      <property name="text">
-      <string>Geometry WKT:</string>
+      <string>Or Enter Geometry WKT (as avalid LineString / LineStringZ / LineStringM / LineStringZM) </string>
```
Comment (Gugym) [inline edr_plugin/gui/ui/query_corridor.ui:255]: This needs to change to be Or Enter Geometry WKT (as **a valid** LineString / LineStringZ / LineStringM / LineStringZM)

**Review by Gugym — APPROVED** [review]

**Review by Gugym — CHANGES_REQUESTED** [review]
Please change the minor typo as noted on the code line.

File: edr_plugin/gui/query_tools.py [hunk 2]
```diff
@@ -523,9 +523,22 @@ def setup_data_query_tool(self):
         else:
             self.output_crs = QgsCoordinateReferenceSystem.fromOgcWmsCrs(crs_name)
         corridor_query_data = self.edr_dialog.query_cbo.currentData()
-        width_units = corridor_query_data["link"]["variables"]["width-units"]
+        variables = corridor_query_data["link"]["variables"]
+        if "width-units" not in variables:
+            self.edr_dialog.plugin.communication.show_error(
+                "Collection does not have information about `width-units`. The collection is not up with specification."
```
Comment (Gugym) [inline edr_plugin/gui/query_tools.py:531]: Please can this text be changed as follows: "The collection does not have information about `width-units` so is not compatible with the OGC API specification."

Comment (Gugym) [inline edr_plugin/gui/query_tools.py:531]: Please can this text be changed to:  "The collection does not have information about `height-units` so is not compatible with the OGC API specification."

**Review by Gugym — CHANGES_REQUESTED** [review]
Please adjust the text comments as shown.

**Review by Gugym — APPROVED** [review]

**Review by Gugym — CHANGES_REQUESTED** [review]
Please change the minor typo as noted on the code line.

**Comment by JanCaha** [comment]
All changed as required.

**Review by Gugym — APPROVED** [review]

**Pull request merged**